src/main/scala/io/iohk/ethereum/transactions/TransactionHistoryService.scala

```diff
@@ -0,0 +1,143 @@
+package io.iohk.ethereum.transactions
+
+import akka.actor.ActorRef
+import akka.util.Timeout
+import cats.implicits._
+import io.iohk.ethereum.domain._
+import io.iohk.ethereum.jsonrpc.AkkaTaskOps.TaskActorOps
+import io.iohk.ethereum.transactions.PendingTransactionsManager.PendingTransaction
+import io.iohk.ethereum.transactions.TransactionHistoryService.{
+  ExtendedTransactionData,
+  MinedTxChecker,
+  PendingTxChecker
+}
+import io.iohk.ethereum.utils.Logger
+import monix.eval.Task
+import monix.reactive.{Observable, OverflowStrategy}
+
+import scala.collection.immutable.NumericRange
+import scala.concurrent.duration.FiniteDuration
+
+class TransactionHistoryService(
+    blockchain: Blockchain,
+    pendingTransactionsManager: ActorRef,
+    getTransactionFromPoolTimeout: FiniteDuration
+) extends Logger {
+  def getAccountTransactions(
+      account: Address,
+      fromBlocks: NumericRange[BigInt]
+  ): Task[List[ExtendedTransactionData]] = {
+    val txnsFromBlocks = Observable
+      .from(fromBlocks.reverse)
+      .mapParallelOrdered(10)(blockNr => Task { blockchain.getBlockByNumber(blockNr) })(OverflowStrategy.Unbounded)
+      .collect { case Some(block) => block }
+      .concatMap { block =>
+        val getBlockReceipts = Task {
+          blockchain.getReceiptsByHash(block.hash).map(_.toVector).getOrElse(Vector.empty)
+        }.memoizeOnSuccess
```

Comment:
`memoizeOnSuccess` won't work here

Reply:
it will work on a per-block basis, won't it?

Reply:
You're right. I missed it